src/heightmap.cpp

```diff
@@ -146,8 +187,7 @@ static bool ReadHeightmapPNG(const char *filename, uint *x, uint *y, byte **map)
 	uint width = png_get_image_width(png_ptr, info_ptr);
 	uint height = png_get_image_height(png_ptr, info_ptr);
 
-	/* Check if image dimensions don't overflow a size_t to avoid memory corruption. */
-	if ((uint64)width * height >= (size_t)-1) {
+	if (!IsValidHeightmapDimension(width, height)) {
```

Comment:
I think `bytes_per_pixel = 1` is underestimating the usage for PNGs. Maybe default to 3, to get it a bit more realistic?

Reply:
I've just removed the logic for "bytes" and limited pixels to 256M instead of potentially allowing 512M when using grayscale.